File: Excepciones/Raiz.java
import java.lang.ArrayIndexOutOfBoundsException;
import java.lang.NullPointerException;
import java.lang.NumberFormatException;

public class Raiz {

    public static void main (String[] args) {
        boolean error = false;
        Float numero = null;
        try {
            float val = Float.valueOf(args[0]);
            if (val >= 0) numero = val;
            System.out.println("La raíz cuadrada de " + numero
                + " es " + Math.sqrt(numero));
        } catch (NullPointerException excepcion) {
            System.out.println("No puedo calcular la raíz cuadrada"
                + " de un número negativo");
            error = true;
        } catch (ArrayIndexOutOfBoundsException excepcion) {
            System.out.println("Se requiere un número para calcular "
                + "su raíz cuadrada");
            error = true;
        } catch (NumberFormatException excepcion) {
            System.out.println("Error en el valor: " + excepcion);
            error = true;
        } finally {
            if (!error) 
                System.out.println("Bien hecho");
            else
                System.out.println("Inténtalo de nuevo");
        }
    }
}

File: Excepciones/raiz.py
import sys

def main(val):
    error = True
    
    try:
        numero = float(val)
        print("La raíz cuadrada de %f es %f"
            % (numero, numero ** 0.5))
    except TypeError as excepcion:
        print("Tipo de datos incorrecto:", excepcion)
    except ValueError as excepcion:
        print("Error en el valor:", excepcion)
    else:
        print("Sin errores")
        error = False
    finally:
        if not error:
            print("Bien hecho")
        else:
            print("Inténtalo de nuevo")

main(sys.argv[1])
